test/smart-wallet/testGroup1.ts

```diff
@@ -773,4 +775,131 @@ describe("Base Wallet Functionality", function () {
     const tokenBalanceAfter = await token.balanceOf(bob);
     console.log(tokenBalanceAfter.toString());
   });
+
+  // keep mixedAuth for setOwner() and write test cases to update it from owner and from sample social recovery module.
+  it("can update owner by current owner or social recovery module", async function () {
+    /// 1. update setOwner() by social recovery module
+    // current owner - owner accounts[0]
+    const currentOwner = await userSCW.owner();
+    expect(currentOwner).to.equal(accounts[0].address);
+
+    // social recovery module deploy - socialRecoveryModule
+    const SocialRecoveryModule = await ethers.getContractFactory(
+      "SocialRecoveryModule"
+    );
+    socialRecoveryModule = await SocialRecoveryModule.connect(
+      accounts[0]
+    ).deploy();
+    console.log("SocialModule deployed at ", socialRecoveryModule.address);
+
+    // setup social recovery module, set bob, charlie as friends and set threshold as 2
+    await socialRecoveryModule
+      .connect(accounts[0])
+      .setup([bob, charlie], 2, userSCW.address);
```

Comment:
this should probably be only smart account transaction. check candide contract for social recovery and transaction flow

Reply:
updated!